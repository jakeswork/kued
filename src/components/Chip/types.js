import { shape, bool, node } from 'prop-types';

export const propTypes = {
  classes: shape({}).isRequired,
  green: bool,
  red: bool,
  amber: bool,
  tooltip: node,
};

export const defaultProps = {
  green: false,
  red: false,
  amber: false,
  tooltip: null,
};
