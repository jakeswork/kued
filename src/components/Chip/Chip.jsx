import React, { Component } from 'react';

import { propTypes, defaultProps } from './types';
import Text from '../Text';
import Tooltip from '../Tooltip';

class Chip extends Component {
  static propTypes = propTypes;

  static defaultProps = defaultProps;

  state = {
    isHovered: false,
  };

  render() {
    const { isHovered } = this.state;
    const { classes, children, tooltip } = this.props;

    return (
      <div
        data-test-id="chip-root"
        className={classes.root}
        onMouseEnter={() => this.setState({ isHovered: true })}
        onMouseLeave={() => this.setState({ isHovered: false })}
      >
        <Text caption bold style={{ color: 'white' }}>{ children }</Text>
        {
          tooltip && (
            <Tooltip visible={isHovered} data-test-id="tooltip">
              { tooltip }
            </Tooltip>
          )
        }
      </div>
    );
  }
}

export default Chip;
