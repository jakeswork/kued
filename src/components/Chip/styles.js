export default theme => ({
  root: {
    padding: '8px 16px',
    display: 'inline-block',
    margin: 4,
    borderRadius: 24,
    position: 'relative',
    fontSize: 12,
    cursor: 'default',
    userSelect: 'none',
    backgroundColor: (props) => {
      if (props.green) return theme.colorGreen;

      if (props.red) return theme.colorRed;

      if (props.amber) return theme.colorAmber;

      return theme.colorBlue;
    },
  },
});
