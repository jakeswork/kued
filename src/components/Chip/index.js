import withStyles from 'react-jss';

import Chip from './Chip';
import styles from './styles';

export default withStyles(styles)(Chip);
