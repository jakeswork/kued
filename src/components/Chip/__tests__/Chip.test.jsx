import React from 'react';
import { shallow } from 'enzyme';

import Chip from '../Chip';

const defaultProps = {
  classes: {},
};

describe('The Chip instance', () => {
  describe('When called with default props', () => {
    it('should render without throwing errors', () => {
      const wrapper = shallow(<Chip {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When hovering over the chip', () => {
    it('should render a tooltip', () => {
      const newProps = {
        ...defaultProps,
        tooltip: 'test',
      };
      const wrapper = shallow(<Chip {...newProps} />);
      const chip = wrapper.find({ 'data-test-id': 'chip-root' });

      expect(wrapper.state('isHovered')).toBe(false);

      chip.simulate('mouseenter');

      expect(wrapper.state('isHovered')).toBe(true);

      const tooltip = wrapper.find({ 'data-test-id': 'tooltip' });

      expect(tooltip.exists()).toBe(true);

      chip.simulate('mouseleave');

      expect(wrapper.state('isHovered')).toBe(false);
    });
  });
});
