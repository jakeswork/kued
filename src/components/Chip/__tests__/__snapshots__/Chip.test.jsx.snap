// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Chip instance When called with default props should render without throwing errors 1`] = `
<div
  data-test-id="chip-root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Jss(Text)
    bold={true}
    caption={true}
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    link={null}
    style={
      Object {
        "color": "white",
      }
    }
  />
</div>
`;
